spec.html

```diff
@@ -7137,6 +7193,251 @@ <h1>
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[DisposeMethod]]
+            </td>
+            <td>
+              a function object or *undefined*
+            </td>
+            <td>
+              A function object that will be called with [[ResourceValue]] as its *this* value when the resource disposed.
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id="sec-newdisposecapability" type="abstract operation">
+      <h1>NewDisposeCapability ( ): a DisposeCapability Record</h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _stack_ be a new empty List.
+        1. Return the DisposeCapability Record { [[DisposableResourceStack]]: _stack_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-adddisposableresource" type="abstract operation">
+      <h1>
+        AddDisposableResource (
+          _disposeCapability_: a DisposeCapability Record,
+          _V_: an ECMAScript language value,
+          _hint_: either ~sync-dispose~ or ~async-dispose~,
+          optional _method_: a function object,
+        ): either a normal completion containing ~unused~ or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _method_ is not present, then
+          1. If _V_ is either *null* or *undefined* and _hint_ is ~sync-dispose~, return ~unused~.
+          1. NOTE: When _V_ is either *null* or *undefined* and _hint_ is ~async-dispose~, we record that the resource was evaluated to ensure we will still perform an Await when resources are later disposed.
+          1. Let _resource_ be ? CreateDisposableResource(_V_, _hint_).
+        1. Else,
+          1. Assert: V is *undefined*.
+          1. Let _resource_ be ? CreateDisposableResource(*undefined*, _hint_, _method_).
+        1. Append _resource_ to _disposeCapability_.[[DisposableResourceStack]].
+        1. Return ~unused~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdisposableresource" type="abstract operation">
+      <h1>
+        CreateDisposableResource (
+          _V_: an ECMAScript language value,
+          _hint_: either ~sync-dispose~ or ~async-dispose~,
+          optional _method_: a function object,
+        ): either a normal completion containing a DisposableResource Record or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _method_ is not present, then
+          1. If _V_ is either *null* or *undefined*, then
+            1. Set _V_ to *undefined*.
+            1. Set _method_ to *undefined*.
+          1. Else,
+            1. If _V_ is not an Object, throw a *TypeError* exception.
+            1. Set _method_ to ? GetDisposeMethod(_V_, _hint_).
+            1. If _method_ is *undefined*, throw a *TypeError* exception.
```

Comment:
Why do it only check for `undefined` and not `isCallable`?

Reply:
GetDisposeMethod calls [GetMethod](https://tc39.es/ecma262/#sec-getmethod), which does a IsCallable check.